code/builders/builder-vite/src/build.ts

```diff
@@ -28,5 +37,20 @@ export async function build(options: Options) {
   }).build;
 
   const finalConfig = await presets.apply('viteFinal', config, options);
+
+  const turbosnapPluginName = 'rollup-plugin-turbosnap';
+  const hasTurbosnapPlugin =
+    finalConfig.plugins && hasVitePlugins(finalConfig.plugins, [turbosnapPluginName]);
```

Comment:
It seems that this code is causing a behavior that always displays a warning.

```suggestion
    finalConfig.plugins && await hasVitePlugins(finalConfig.plugins, [turbosnapPluginName]);
```

Reply:
Hi @acid-chicken -- that makes sense! But how to reproduce? I'm not seeing the problem on a basic `react-vite` `8.0.0-beta.2` sandbox project. Any tips?

Reply:
<img width="905" alt="image" src="https://github.com/storybookjs/storybook/assets/132554/56a4e46a-c97b-47a9-b357-1ef9a32918c2">

Reply:
@tmeasday Thank you for the reply!

Digging a little deeper into this issue, I found that this log only appears at build time.
Therefore, you can probably reproduce by executing `yarn create vite` (w/ react & typescript option), `yarn dlx storybook@next init`, then `yarn build-storybook` (not `yarn storybook`).

I hope this will be of some help.

Reply:
Ahh, OK of course. Thank you @acid-chicken